lib/BoundaryValueDiffEqFIRK/src/firk.jl

```diff
@@ -314,19 +350,28 @@ function SciMLBase.solve!(cache::FIRKCacheExpand{iip, T}) where {iip, T}
         end
     end
 
-    u = shrink_y(
-        [reshape(y, cache.in_size) for y in cache.y₀], length(cache.mesh), cache.stage)
+    # Parameter estimation, put the estimated parameters to sol.prob.p
+    if fit_parameters
+        length_u = cache.M - length(prob.p)
+        prob = remake(prob; p = first(cache.y₀)[(length_u + 1):end])
+        map(x -> resize!(x, length_u), cache.y₀)
+        resize!(cache.fᵢ₂_cache, length_u)
+    end
+
```

Comment:
We'll need to support the SciMLStructures interface, but assuming `p` is a vector is fine for now.